src/include/mallocMC/creationPolicies/Scatter_impl.hpp

```diff
@@ -716,14 +711,15 @@ namespace ScatterKernelDetail{
       {
         uint32 linid = threadIdx.x + blockDim.x*(threadIdx.y + threadIdx.z*blockDim.y);
         uint32 threads = blockDim.x*blockDim.y*blockDim.z;
-        uint32 linblockid = blockIdx.x + gridDim.x*(blockIdx.y + blockIdx.z*gridDim.y);
+        const uint32 linblockid = blockIdx.x + gridDim.x*(blockIdx.y + blockIdx.z*gridDim.y);
         uint32 blocks =  gridDim.x*gridDim.y*gridDim.z;
-        linid = linid + linblockid*threads;
+        linid += linblockid*threads;
+
+        uint32 numregions = ((unsigned long long)memsize)/( ((unsigned long long)regionsize)*(sizeof(PTE)+pagesize)+sizeof(uint32)); // TODO(bgruber): why unsigned long long and not size_t?
```

Comment:
I think we want to be sure that this is 64bit and not overflow. `size_t` is on 32bit architectures 4 byte.

Reply:
Hmm, firstly: Do you encounter 32bit system because CUDA is a different architecture than the host and could be 32bits? 

Secondly, assuming the RHS of the divison could be larger than the domain of size_t and thus be larger than the maximum value of the LHS, which is of type size_t, the result would be 0. Thus numregions, and numpages would be 0. And then I guess the whole allocator would not work?
So if the RHS can never assume a value greater than the maximum size_t can hold without loosing its meaning, the computation can be done in size_t.

Since the RHS is the size of a memory region, one could also argue that we should assert that it is not larger than the entire allocated pool (ie. memsize).

What do you think?

Reply:
We used PIConGPU with mallocMC few years ago on a jetson (32bit). Out of the box I can not say if 32bit is still required. I would keep it open and shift this designation to a later PR.

Reply:
I see. I will remove the TODO then and let's keep the code as is.